kyverno/kyverno
# [Improvement] Kyverno should not delete downstream resources when a generate policy using the clone behavior has synchronize: true

## Related issue
closes https://github.com/kyverno/kyverno/issues/1401


## What type of PR is this
> /kind feature

## Proposed Changes
- Skipping deletion of generate request for deletion of generate policy with clone.
- After the clone policy with sync=true is deleted, Kyverno will no longer maintain the generated resource. Any changes (UPDATE,DELETE) to the source won't be synced to the generate resource.
- policy name is removed from the source resource label `generate.kyverno.io/clone-policy-name`

### Proof Manifests
1. apply ConfigMap:
```
 apiVersion: v1
kind: ConfigMap
metadata:
  name: corp-ca-cert
  namespace: default
data:
  ca: |-
    -----BEGIN CERTIFICATE-----
    MIID5zCCAs+gAwIBAgIUCl6BKlpe2QiS5IQby6QOW7vexMwwDQYJKoZIhvcNAQEL
    BQAwgYIxCzAJBgNVBAYTAlVTMQswCQYDVQQIDAJDQTENMAsGA1UEBwwEVG93bjEQ
    MA4GA1UECgwHQ29tcGFueTENMAsGA1UECwwEQ29ycDEYMBYGA1UEAwwPY29ycC5k
    Rx0PftMOuWpaV0UtodJkOjXfE+hMeHXbkunUZVkIB8N9VJfouFQyBAurmw======
    -----END CERTIFICATE-----
```

2. apply policy:
```
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: generate-policy
spec:
  rules:
  - name: copy-corp-ca-cert
    match:
      resources:
        kinds:
        - Namespace
        selector:
          matchLabels:
            app-type: corp
    exclude:
      resources:
        namespaces:
        - kube-system
        - default
        - kube-public
        - kyverno
    generate:
      kind: ConfigMap
      name: corp-ca-cert
      namespace: "{{request.object.metadata.name}}"
      synchronize: true
      clone:
        namespace: default
        name: corp-ca-cert
```

3. apply namespace:
```
apiVersion: v1
kind: Namespace
metadata:
  name: test1
  labels:
    app-type: corp
```

4. check ConfigMap in namespace `test1`
```
NAMESPACE     NAME              
test1       corp-ca-cert              
```

5. delete the applied policy and then check the ConfigMap in `test1` namespace. The configMap should be present in the namespace even after deletion of the policy. 


## Checklist

- [] I have added tests that prove my fix is effective or that my feature works - https://github.com/kyverno/kyverno/issues/2085
  - [] I have added or changed [the documentation](https://github.com/kyverno/website) myself in an existing PR and the link is:
  
  
  - [X] I have raised an issue in [kyverno/website](https://github.com/kyverno/website) to track the doc update and the link is:
 https://github.com/kyverno/website/issues/195
  - [] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.
